### DISCUSSION TITLE
Lets add "enabled/disabled" features per project.

### DISCUSSION THREAD

**Discussion by belonesox** [sagemathinc/cocalc#6494] [Ideas]
At the moment, each project detects available capabilities on startup (`cocalc/src/packages/project/configuration.ts`).
But on the same server (or on the same docker image) there can be 
* projects for "experienced users" (using all the capabilities of cocalc)
* Projects for students — there UI must be clear and simple, not like Bugzilla/EnterpriseApp ( [illustration](https://rodgersnotes.files.wordpress.com/2010/10/simplicity.png) ). 

Very hard choice!

![image](https://user-images.githubusercontent.com/1609739/221692271-4411ec32-705e-45f3-a137-f54b687c3b0e.png)

I see the last UI with square buttons looks a bit more estetic, but actually anyway we lost a lot of space for something weird and rarely used.

![image](https://user-images.githubusercontent.com/1609739/221694245-60bd0515-30d8-4160-9186-d78548c703ea.png)

* 1 sq feet for 
![image](https://user-images.githubusercontent.com/1609739/221692734-985d5421-4b74-4947-aea1-782fd978e939.png) — if «access blocked» —«why are your here?»  And I failed to find correspongin project settings.  

* «Timers» and «Todo list» — hmm.... they used really rarely. 
*  «Graphical desktop» — I still failed to make installation with acceptable latency, may be someday Antoine improves Xpra, and it can be usable. 
* Now we need to scroll for «Upload» drag-and-drop pad.

Let stop arguing what artefact is important, just make it tunable, for example, 
for some student «Jupyter Notebook» + «Folder» will be sufficient, for old R-lovers «Folder + RMarkdown» will be OK, etc.

So, lets «enable disable» available features on project settings page — for example, just add additional checkbox per row on «avalable features» block, that enabled (or disabled) feature per project + fea. Of course, somehow store it on DB (JSON?), etc…

**Comment by williamstein** [comment]
In course configuration you can disable/enable a large number of things across all the projects in a course:

<img width="781" alt="image" src="https://user-images.githubusercontent.com/1276278/221700815-7e39c734-46f1-4363-bb25-fa05a3c6cfa3.png">

This could (does?) change what buttons appear in +New.  This could be expanded to impact more of what is available there.
